lib/httplog/http_log.rb

```diff
@@ -197,6 +199,48 @@ def colorize(msg)
 
     private
 
+    def masked(msg, key=nil)
+      return msg if config.filter_parameters.empty?
+      return msg if msg.nil?
+
+      # If a key is given, msg is just the value and can be replaced
+      # in its entirety.
+      return (config.filter_parameters.include?(key.downcase) ? PARAM_MASK : msg) if key
+
+      # Otherwise, we'll parse Strings for key=valye pairs...
+      case msg
+      when *string_classes
+        config.filter_parameters.reduce(msg) do |m,key|
+          m.to_s.gsub(/(#{key})=[^&]+/i, "#{key}=#{PARAM_MASK}")
```

Comment:
This is a bit more robust, just `\w+` would only replace up to the first possible whitespace, URL encoded or not. Not terribly pretty but is might suffice and is less complex than parsing the full query string into a hash.